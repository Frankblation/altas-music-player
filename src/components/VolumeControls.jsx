import React from "react"

export default function VolumeControls({ volume, onVolumeChange }) {
  return (
    <div className="flex items-center space-x-2 mt-4">
      <svg
        xmlns="http://www.w3.org/2000/svg"
        className="h-6 w-6 text-gray-600"
        fill="none"
        viewBox="0 0 24 24"
        stroke="currentColor"
      >
        <path
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth={2}
          d="M15.536 8.464a5 5 0 010 7.072m2.828-9.9a9 9 0 010 12.728M5.586 15H4a1 1 0 01-1-1v-4a1 1 0 011-1h1.586l4.707-4.707C10.923 3.663 12 4.109 12 5v14c0 .891-1.077 1.337-1.707.707L5.586 15z"
        />
      </svg>
      <input
        type="range"
        className="w-full"
        min="0"
        max="100"
        value={volume}
        onChange={(e) => onVolumeChange(Number.parseInt(e.target.value))}
      />
    </div>
  )
}

