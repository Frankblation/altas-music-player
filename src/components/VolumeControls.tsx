import { useState } from "react";

const VolumeControls = ({ volume, setVolume }) => {
  const handleVolumeChange = (e) => {
    const newVolume = parseFloat(e.target.value); // Get value as float
    setVolume(newVolume); // Update parent component state
  };

  return (
    <div className="flex flex-col items-center space-y-2">
      <input
        type="range"
        min="0"
        max="1"
        step="0.01"
        value={volume}
        onChange={handleVolumeChange}
        className="w-32 h-2 bg-blue-400 rounded-lg appearance-none focus:outline-none focus:ring-2 focus:ring-blue-300"
      />
      <div className="text-gray-600 text-sm">
        Volume: {(volume * 100).toFixed(0)}%
      </div>
    </div>
  );
};

export default VolumeControls;
