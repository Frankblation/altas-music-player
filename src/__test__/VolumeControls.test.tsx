import { render } from "@testing-library/react";
import { expect, test, vi } from "vitest";
import VolumeControls from "../components/VolumeControls";

test("VolumeControls renders with default volume", () => {
  const mockSetVolume = vi.fn();
  const { container } = render(
    <VolumeControls volume={0.5} setVolume={mockSetVolume} />
  );
  expect(container).toMatchSnapshot();
});
