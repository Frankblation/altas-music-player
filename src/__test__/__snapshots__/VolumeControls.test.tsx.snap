// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`VolumeControls renders with default volume 1`] = `
<div>
  <div
    class="flex flex-col items-center space-y-2"
  >
    <input
      class="w-32 h-2 bg-blue-400 rounded-lg appearance-none focus:outline-none focus:ring-2 focus:ring-blue-300"
      max="1"
      min="0"
      step="0.01"
      type="range"
      value="0.5"
    />
    <div
      class="text-gray-600 text-sm"
    >
      Volume: 
      50
      %
    </div>
  </div>
</div>
`;
